test/hotspot/gtest/compiler/test_directivesParser.cpp

```diff
@@ -31,15 +31,16 @@
 
 class DirectivesParserTest : public ::testing::Test{
  protected:
-  const char* const _locale;
+  char* const _locale;
   ResourceMark rm;
   stringStream stream;
   // These tests require the "C" locale to correctly parse decimal values
-  DirectivesParserTest() : _locale(setlocale(LC_NUMERIC, nullptr)) {
+  DirectivesParserTest() : _locale(os::strdup(setlocale(LC_NUMERIC, nullptr), mtTest)) {
     setlocale(LC_NUMERIC, "C");
```

Comment:
Would it fix the issue if we did this instead?

```suggestion
  DirectivesParserTest() : _locale(setlocale(LC_NUMERIC, "C")) {
```

seems to me that the string returned by setlocale is only valid until the next setlocale call, and currently we call setlocale twice in the constructor, and save the result of the first call.

Reply:
No. The first setlocate call returns the pointer to the last locale, which becomes invalid. Changing the input string on the first setlocale call won't change that.

Reply:
Ah. I was misled by the `setlocale` docs:
> The string returned is such that a subsequent call with that string and its associated category will restore that part of the process's locale.

apparently it doesn't restore them _to the previous value_, as I incorrectly assumed.